RD-3190 Moved tenant selection from top to side bar

kubama opened the pull request:
<!--- Provide JIRA issue ID and a general summary of your changes in the Title above -->

## Description
Introduced `SystemMenu` component, extracted and reused `SideBarItem` component.
<!--- Describe your changes to help reviewers understand the details. -->

## Screenshots / Videos
![image](https://user-images.githubusercontent.com/49753069/140921755-e24f9387-40e3-4f0a-a56c-c6cb274d3767.png)

<!--- If applicable, add screenshot(s) or video(s) describing the changes you made. -->
<!--- Consider adding comparison screenshot(s) - before and after the change. -->
<!--- If not applicable, just write “N/A” in this section. -->

## Checklist
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
- [X] My code follows the UI Code Style.
- [x] I verified that all tests and checks have passed.
- [X]  I verified if that change requires a documentation update.
- [X] I added proper labels to this PR.

## Tests
https://jenkins.cloudify.co/view/UI/job/Stage-UI-System-Test/1467/
<!--- If applicable, describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
<!--- If applicable, provide a link to system tests, e.g: -->
<!--- [#1039](https://jenkins.cloudify.co/.../1039) (2021-08-06 07:05:50) --> 
<!--- If not applicable, just write “N/A” in this section. -->

## Documentation
Will be covered in separate ticket
<!--- If applicable, describe how you documented or plan to document your changes. -->
<!--- Provide links to JIRA issues, other PRs or related documentation pages. -->
<!--- If not applicable, just write “N/A” in this section. -->

qooban commented on test/cypress/components/app/sidebar/TenantSelection_spec.tsx:
```diff
@@ -0,0 +1,68 @@
+import TenantSelection from 'app/components/sidebar/TenantSelection';
+import { mountWithProvider } from '../../utils';
+
+describe('TenantSelection', () => {
```
General comment about test case names - I see they were moved from the Enzyme test, but maybe it'd be worth to update it a bit? E.g. "TenantSelection renders tenants list has selected" doesn't seem to be valid sentence.

kubama replied:
Right, I improved the wording

qooban commented on app/components/IconSelection.tsx:
```diff
@@ -7,7 +7,7 @@ import { useInput, useOpen } from '../utils/hooks';
     style?: CSSProperties;
     value?: SemanticICONS;
-    onChange: (value?: SemanticICONS) => void;
+    onChange?: (value?: SemanticICONS) => void;
```
Why we need that change?

kubama replied:
When `enabled` prop is statically set to `false` then `onChange` prop is not needed.

qooban replied:
👍🏻

qooban commented on app/components/sidebar/SystemMenu.tsx:
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import TenantSelection from './TenantSelection';
+
+const SystemMenu: FunctionComponent = () => {
```
I thought originally about plural - `SystemMenus` as there'll be more menus in that area - User Menu, Help Menu.
I'm not against, I'm just interested in what was your reasoning?

Actually, as User Menu and Help Menu will be a group item after all, it may make more sense to have singular version.

kubama replied:
I see it as a single menu containing submenus rather than a collection of independent menus, thus I think singular form suits better.

qooban replied:
👍🏻

qooban commented on app/components/sidebar/TenantSelection.tsx:
```diff
@@ -0,0 +1,84 @@
+    return (
+        <Dropdown trigger={tenantMenuTrigger} onClose={clearSearch} pointing="left" icon={null} fluid>
+            <Dropdown.Menu>
+                <Input
```
Optional:

I found that there's also `Dropdown.SearchInput` component in SUIR, maybe it can be used here instead of generic `Input`?

kubama replied:
I gave it a try but it's barely usable. It lacks regular `Input` styling as well as most `Input` props.

qooban reviewed (commented):
Looks good 👍🏻 Thanks for keeping this PR small.
I assume that there's something still to do with system tests.

Did some UI tests and it seems working fine. Some questions:
1. What is your impression with search input changing place (when typing)? Should we keep it on top as is? Did you test having it on the bottom?
2. As tenant selector shows currently selected tenant as menu item label, I think we should add some help for the user to describe what is it. There's neither a tooltip on hovering the item nor a header inside the popup. What do you think about improving it?

kubama commented on app/components/layout/Header.tsx:
```diff
@@ -70,7 +69,7 @@ export default class Header extends Component {
                         </div>
                     )}
-                    {this.isModeMain() && <Tenants manager={manager} />}
+                    {/* {this.isModeMain() && <Tenants manager={manager} />} */}
```
Of course, removed

kubama commented on app/components/sidebar/SideBar.tsx:
```diff
@@ -33,6 +38,7 @@ const SideBar: FunctionComponent<SideBarProps> = ({ pageId }) => {
         <div className="sidebarContainer">
             <ThemedSidebar theme={theme} visible className={`vertical menu small basic ${className}`}>
```
Looks like the `Sidebar` component doesn't have the corresponding props. I only removed the `basic` class as it seemed to have no effect.

qooban replied:
I'm wondering how it works. Maybe `as` prop should be set to `Menu` and the rest of the options are props passed to `Menu`. Anyways, let's leave it. Maybe during RD-3482 we can remove some of those classes.

kubama commented:
> 1. What is your impression with search input changing place (when typing)? Should we keep it on top as is? Did you test having it on the bottom?

I moved it to bottom and tested it - I can see no issues, so I think it's a more suitable placement.

> 2. As tenant selector shows currently selected tenant as menu item label, I think we should add some help for the user to describe what is it. There's neither a tooltip on hovering the item nor a header inside the popup. What do you think about improving it?

I added dropdown header - I think it does the job.

kubama commented:
I submitted system tests execution after providing test fixes - if it passes then I'm not planning any more commits.

qooban reviewed (dismissed):
Looks good 👍🏻

qooban commented on test/cypress/components/app/sidebar/PagesList_spec.tsx:
```diff
@@ -73,7 +73,7 @@ describe('PagesList', () => {
             cy.contains('.item', 'Page_Group_0').find('.edit').click({ force: true });
             cy.get('input').type('2{enter}');
-            cy.contains('.item', 'Page_Group_0').find('.expand').click();
+            cy.contains('.item', 'Page_Group_0').find('.expand').click({ force: true });
```
Is the `force` set because the expand icon is covered by something or not in the view?

kubama replied:
Yes, it's partially out of the view. This is because in the test the component is rendered out of the context in which it is rendered in the real application.

qooban replied:
Got it 👍🏻

qooban reviewed (approved):
LGTM 👍🏻
